src/gentropy/dataset/l2g_feature.py

```diff
@@ -395,95 +768,471 @@ def compute(
+            feature_dependency (dict[str, Any]): Dataset with the colocalisation results
 
         Returns:
-                L2GFeature: Feature dataset
-
-        Raises:
-            NotImplementedError: Not implemented
+            TuQtlColocH4MaximumNeighbourhoodFeature: Feature dataset
         """
-        raise NotImplementedError
+        colocalisation_method = "Coloc"
+        colocalisation_metric = "h4"
+        qtl_type = "tuqtl"
+        return cls(
+            _df=convert_from_wide_to_long(
+                _common_colocalisation_feature_logic(
+                    study_loci_to_annotate,
+                    colocalisation_method,
+                    colocalisation_metric,
+                    cls.feature_name,
+                    qtl_type,
+                    **feature_dependency,
+                ),
+                id_vars=("studyLocusId", "geneId"),
+                var_name="featureName",
+                value_name="featureValue",
+            ),
+            _schema=cls.get_schema(),
+        )
 
 
-class DistanceTssMeanFeature(L2GFeature):
-    """Average distance of all tagging variants to gene TSS.
+def _common_distance_feature_logic(
+    study_loci_to_annotate: StudyLocus | L2GGoldStandard,
+    *,
+    variant_index: VariantIndex,
+    feature_name: str,
+    distance_type: str,
+    agg_expr: Column,
+) -> DataFrame:
+    """Computes the feature.
 
-    NOTE: to be rewritten taking variant index as input
+    Args:
+        study_loci_to_annotate (StudyLocus | L2GGoldStandard): The dataset containing study loci that will be used for annotation
+        variant_index (VariantIndex): The dataset containing distance to gene information
+        feature_name (str): The name of the feature
+        distance_type (str): The type of distance to gene
+        agg_expr (Column): The expression that aggregate distances into a specific way to define the feature
+
+    Returns:
+            DataFrame: Feature dataset
     """
+    distances_dataset = variant_index.get_distance_to_gene(distance_type=distance_type)
+    return (
+        study_loci_to_annotate.df.withColumn("variantInLocus", f.explode_outer("locus"))
+        .select(
+            "studyLocusId",
+            f.col("variantInLocus.variantId").alias("variantInLocusId"),
+            f.col("variantInLocus.posteriorProbability").alias(
+                "variantInLocusPosteriorProbability"
+            ),
+        )
+        .join(
+            distances_dataset.withColumnRenamed(
+                "variantId", "variantInLocusId"
+            ).withColumnRenamed("targetId", "geneId"),
+            on="variantInLocusId",
+            how="inner",
+        )
+        .withColumn(
+            "weightedDistance",
+            f.col(distance_type) * f.col("variantInLocusPosteriorProbability"),
+        )
+        .groupBy("studyLocusId", "geneId")
+        .agg(agg_expr.alias(feature_name))
+    )
+
```

Comment:
Are we using weighted distances as the feature now? instead of the normalised distance score between 0-1?

Reply:
We used to weight them earlier as well. The feature itself is not normalised until later.